homeassistant/components/switchgrid/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""The Switchgrid integration."""
+
+from __future__ import annotations
+
+from switchgrid_python_client import SwitchgridData
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+from .coordinator import SwitchgridCoordinator
+
+PLATFORMS: list[Platform] = [Platform.CALENDAR]
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Switchgrid from YAML configuration."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Switchgrid from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    session = async_get_clientsession(hass)
+
+    data = SwitchgridData(session)
+
+    coordinator = SwitchgridCoordinator(hass, data)
+    await coordinator.async_config_entry_first_refresh()
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    await coordinator.async_refresh()
```

Comment:
Why do we refresh twice?

Reply:
So, without this I was not seeing the calendar updating on HA restart. But this is because I was initializing the _events prop to an empty array in the calendar platform.

Fixed that, and removed the `async_refresh()`.